### PR TITLE
Bar Chart - bug fix

### PR DESCRIPTION
**Pull request by lilyclements**

### CODE REVIEW

File: instat/dlgBarAndPieChart.Designer.vb [hunk 1]
```diff
@@ -51,7 +51,7 @@ Partial Class dlgBarAndPieChart
         'lblSecondFactor
         '
         Me.lblSecondFactor.AutoSize = True
-        Me.lblSecondFactor.Location = New System.Drawing.Point(254, 120)
+        Me.lblSecondFactor.Location = New System.Drawing.Point(253, 124)
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.Designer.vb]: Sorted location/layout

File: instat/dlgBarAndPieChart.Designer.vb [hunk 2]
```diff
@@ -60,27 +60,27 @@ Partial Class dlgBarAndPieChart
         '
         'cmdBarChartOptions
         '
-        Me.cmdBarChartOptions.Location = New System.Drawing.Point(11, 230)
+        Me.cmdBarChartOptions.Location = New System.Drawing.Point(10, 227)
         Me.cmdBarChartOptions.Name = "cmdBarChartOptions"
         Me.cmdBarChartOptions.Size = New System.Drawing.Size(120, 25)
-        Me.cmdBarChartOptions.TabIndex = 9
+        Me.cmdBarChartOptions.TabIndex = 8
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.Designer.vb]: Tab index addressed

File: instat/dlgBarAndPieChart.Designer.vb [hunk 3]
```diff
@@ -60,27 +60,27 @@ Partial Class dlgBarAndPieChart
...
         'cmdOptions
         '
-        Me.cmdOptions.Location = New System.Drawing.Point(10, 262)
+        Me.cmdOptions.Location = New System.Drawing.Point(10, 258)
         Me.cmdOptions.Name = "cmdOptions"
         Me.cmdOptions.Size = New System.Drawing.Size(120, 25)
         Me.cmdOptions.TabIndex = 10
         Me.cmdOptions.Tag = "Plot_Options..."
-        Me.cmdOptions.Text = "Plot Options..."
+        Me.cmdOptions.Text = "Plot Options"
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.Designer.vb]: Removed ellipsis for consistency

File: instat/dlgBarAndPieChart.Designer.vb [hunk 4]
```diff
@@ -60,27 +60,27 @@ Partial Class dlgBarAndPieChart
...
         Me.cmdOptions.UseVisualStyleBackColor = True
         '
         'cmdPieChartOptions
         '
-        Me.cmdPieChartOptions.Location = New System.Drawing.Point(12, 231)
+        Me.cmdPieChartOptions.Location = New System.Drawing.Point(10, 227)
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.Designer.vb]: both cmdChartOptions (Pie and Bar) are now on top of one another so there is no "jig" when switching rdos

File: instat/dlgBarAndPieChart.Designer.vb [hunk 5]
```diff
@@ -126,23 +126,23 @@ Partial Class dlgBarAndPieChart
         '
         'ucrSaveBar
         '
-        Me.ucrSaveBar.Location = New System.Drawing.Point(10, 306)
+        Me.ucrSaveBar.Location = New System.Drawing.Point(10, 315)
         Me.ucrSaveBar.Name = "ucrSaveBar"
-        Me.ucrSaveBar.Size = New System.Drawing.Size(313, 24)
-        Me.ucrSaveBar.TabIndex = 11
+        Me.ucrSaveBar.Size = New System.Drawing.Size(255, 24)
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.Designer.vb]: Save bar size at 255, as I've been doing on other graphics dialogs

File: instat/dlgBarAndPieChart.Designer.vb [hunk 6]
```diff
@@ -126,23 +126,23 @@ Partial Class dlgBarAndPieChart
...
-        Me.ucrSaveBar.Size = New System.Drawing.Size(313, 24)
-        Me.ucrSaveBar.TabIndex = 11
+        Me.ucrSaveBar.Size = New System.Drawing.Size(255, 24)
+        Me.ucrSaveBar.TabIndex = 12
         '
         'ucrChkFlipCoordinates
         '
         Me.ucrChkFlipCoordinates.Checked = False
-        Me.ucrChkFlipCoordinates.Location = New System.Drawing.Point(255, 168)
+        Me.ucrChkFlipCoordinates.Location = New System.Drawing.Point(10, 289)
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.Designer.vb]: Moved this to be in the same location as it is for the other graphics dialogs

File: instat/dlgBarAndPieChart.vb [hunk 7]
```diff
@@ -103,6 +104,7 @@ Public Class dlgBarAndPieChart
         ucrBarChartSelector.Reset()
         ucrFactorReceiver.SetMeAsReceiver()
         ucrSaveBar.Reset()
+        SetDialogOptions()
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.vb]: Calling this in SetDefaults() after the `bUserTyped` addition

File: instat/dlgBarAndPieChart.vb [hunk 8]
```diff
@@ -126,7 +128,7 @@ Public Class dlgBarAndPieChart
         ucrBase.clsRsyntax.SetBaseROperator(clsBaseOperator)
     End Sub
 
-    Private Sub SetRCodeForControls(bReset As Boolean)
+    Public Sub SetRCodeForControls(bReset As Boolean)
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.vb]: Public sub

File: instat/dlgBarAndPieChart.vb [hunk 9]
```diff
@@ -189,33 +191,31 @@ Public Class dlgBarAndPieChart
...
-            clsRgeomBarFunction.RemoveParameterByName("width")
-            clsBaseOperator.RemoveParameter(clsRCoordPolarParam)
-        ElseIf rdoPieChart.Checked Then
-            clsRggplotFunction.AddParameter("mapping", clsRFunctionParameter:=clsPieAesFunction, iPosition:=1)
-            ucrSaveBar.SetPrefix("Pie")
-            clsRgeomBarFunction.AddParameter("width", "1")
-            clsBaseOperator.AddParameter(clsRCoordPolarParam)
-            cmdPieChartOptions.Visible = True
-            cmdBarChartOptions.Visible = False
+    Private Sub SetDialogOptions()
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.vb]: Fixed name typo

File: instat/dlgBarAndPieChart.vb [hunk 10]
```diff
@@ -189,33 +191,31 @@ Public Class dlgBarAndPieChart
...
     End Sub
 
     Private Sub ucrPnlOptions_ControlValueChanged() Handles ucrPnlOptions.ControlValueChanged
-        SetDailogOptions()
-    End Sub
-
-    Private Sub AllControls_ContentsChanged() Handles ucrFactorReceiver.ControlContentsChanged, ucrSaveBar.ControlContentsChanged
-        TestOkEnabled()
+        SetDialogOptions()
     End Sub
```
Comment (lilyclements) [inline instat/dlgBarAndPieChart.vb]: This sub was being called twice, removed one of the subs

**Comment by lilyclements** [comment]
@maxwellfundi I have fixed some minor issues in this dialog. On top of this, I have encountered two big issues that need to be fixed. Can you make a pull request of this and fix the following:

* Reset does not reset the radio buttons
* If the receiver is full on the Pie Chart rdo, it crashes if I click on "Pie Chart Options" and then click "return".

The reset problem is also an issue on bar chart from summary.

**Comment by maxwellfundi** [comment]
@lilyclements  i am now on this. You can close this now.

**Pull request closed without merging**